generate_outputs.py

```diff
@@ -0,0 +1,230 @@
+    coded_dir_path = args.coded_dir_path
+    csv_by_message_output_path = args.csv_by_message_output_path
+    csv_by_individual_output_path = args.csv_by_individual_output_path
+    production_csv_output_path = args.production_csv_output_path
+
+    message_paths = [s01e01_input_path, s01e02_input_path, s01e03_input_path,
+                     s01e04_input_path, s01e05_input_path, s01e06_input_path, s01e07_input_path]
+    
+    follow_up_survey_paths = [s01_follow_up_w2_input_path, s01_follow_up_w6_input_path]
+
+    # Load the pipeline configuration file
+    log.info("Loading Pipeline Configuration File...")
+    with open(pipeline_configuration_file_path) as f:
+        pipeline_configuration = PipelineConfiguration.from_configuration_file(f)
+
+    if pipeline_configuration.drive_upload is not None:
+        # Fetch the Rapid Pro Token from the Google Cloud Storage URL
+        parsed_rapid_pro_token_file_url = urlparse(pipeline_configuration.drive_upload.drive_credentials_file_url)
+        bucket_name = parsed_rapid_pro_token_file_url.netloc
+        blob_name = parsed_rapid_pro_token_file_url.path.lstrip("/")
+
+        log.info(f"Downloading Drive service account credentials from file '{blob_name}' in bucket '{bucket_name}'...")
+        storage_client = storage.Client.from_service_account_json(google_cloud_credentials_file_path)
+        credentials_bucket = storage_client.bucket(bucket_name)
+        credentials_blob = credentials_bucket.blob(blob_name)
+        credentials_info = json.loads(credentials_blob.download_as_string())
+        log.info("Downloaded Drive service account credentials")
+
+        drive_client_wrapper.init_client_from_info(credentials_info)
+
+    # Load phone number <-> UUID table
+    log.info("Loading Phone Number <-> UUID Table...")
+    with open(phone_number_uuid_table_path, "r") as f:
+        phone_number_uuid_table = PhoneNumberUuidTable.load(f)
+
+    # Load messages
+    messages_datasets = []
+    for i, path in enumerate(message_paths):
+        log.info("Loading Episode {}/{}...".format(i + 1, len(message_paths)))
+        with open(path, "r") as f:
+            messages_datasets.append(TracedDataJsonIO.import_json_to_traced_data_iterable(f))
+        log.debug(f"Loaded {len(messages_datasets[-1])} messages")
+
+    # Load demogs
+    log.info("Loading Demographics...")
+    with open(s01_demog_input_path, "r") as f:
+        s01_demographics = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+        log.debug(f"Loaded {len(s01_demographics)} contacts")
+    
+    # Load Follow up Surveys
+    survey_datasets = []
+    for i, path in enumerate(follow_up_survey_paths):
+        log.info("Loading Follow up {}/{}...".format(i + 1, len(follow_up_survey_paths)))
+        with open(path, "r") as f:
+            survey_datasets.append(TracedDataJsonIO.import_json_to_traced_data_iterable(f))
+        log.debug(f"Loaded {len(survey_datasets[-1])} messages")
```

Comment:
Think it would be cleaner to assign the new survey to a variable, log the length of that list, then append to survey_datasets.

Reply:
Ah, you got that from me. Sorry, I'm tempted to change ADSS too now